### DISCUSSION TITLE
Plan to upgrade Django to solve PSQL disconnection

### DISCUSSION THREAD

**Discussion by maxware0715** [frePPLe/frepple#643] [General]
Recently my project frequently met the disconnection issue while connect to PSQL server:

django.db.utils.OperationalError: server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.

We investigate that new django version has a setting make liveness check before using connection:
https://docs.djangoproject.com/en/4.1/ref/settings/#conn-health-checks

but it requires django version >= 4.1,

while frepple only record django version=3.2.13:
https://frepple.com/docs/current/release-notes.php#:~:text=Django%20release%20is%20upgraded%20to%203.2.13%20to%20address%20a%20security%20issue.

Is it a good idea to upgrade django in frepple image? 
If yes, any experience of adjustment?
If no, any other workaround method to solve the disconnect issue?

**Comment by jdetaeye** [comment]
Since Frepple 7.0, we use a slightly customized version of django 4.2 (maintained in https://github.com/frePPLe/django).   
Since version 7.0 frepple comes packaged with a virtual python environment with all packages it needs, including django. 

So, it looks like a frepple upgrade to a recent version will resolve your issue.

**Reply by maxware0715** [reply]
Thank you.
Btw, where can I find the django version binding with frepple?

**Reply by jdetaeye** [reply]
You can see all python packages we use in the file https://github.com/frePPLe/frepple/blob/master/requirements.txt    The last one is the link to django.

Normal users shouldn't need to dive this deep.  Just stick to installing the debian package or (even better) the docker container, and don't worry about the internals...

**Reply by maxware0715** [reply]
Well, my team do very much to investigate the root cause of disconnection problem, and still stuck on it...
Currently we try to adjust django setting to prevent it happened again.
Hope we can share the result if we come over it.

**Comment by jdetaeye** [comment]
Just to be clear:   Upgrading django without upgrading frepple is NOT an option.
